stmt/stmts/set.go

```diff
@@ -230,6 +235,30 @@ func (s *SetPwdStmt) SetText(text string) {
 
 // Exec implements the stmt.Statement Exec interface.
 func (s *SetPwdStmt) Exec(ctx context.Context) (_ rset.Recordset, err error) {
-	// TODO: finish this
-	return nil, nil
+	// If len(s.User) == 0, use CURRENT_USER()
+	strs := strings.Split(s.User, "@")
+	userName := strs[0]
+	host := strs[1]
+	// Update mysql.user
+	r := &rsets.JoinRset{
+		Left: &rsets.TableSource{
+			Source: table.Ident{
+				Name:   model.NewCIStr(mysql.UserTable),
+				Schema: model.NewCIStr(mysql.MySQLDB),
+			},
+		},
+	}
+	asgn := expressions.Assignment{
+		ColName: "Password",
+		Expr:    expressions.Value{Val: s.Password},
+	}
+	nameMatch := expressions.NewBinaryOperation(opcode.EQ, &expressions.Ident{CIStr: model.NewCIStr("User")}, &expressions.Value{Val: userName})
+	hostMatch := expressions.NewBinaryOperation(opcode.EQ, &expressions.Ident{CIStr: model.NewCIStr("Host")}, &expressions.Value{Val: host})
```

Comment:
I only mean whether the variable names use nameExpr and hostExpr are better, for nameMatch and nameMatch sound like boolean variables.